### ISSUE TITLE
Web Service Sender does not queue on connection error

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#3358]
Web Service dipatch failes and sets the message status to ERROR if no connection can be made to the external server:

{noformat} 
Web Service Sender error
ERROR MESSAGE: Error creating web service dispatch
java.net.ConnectException: Connection refused
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.PlainSocketImpl.doConnect(PlainSocketImpl.java:351)
	at java.net.PlainSocketImpl.connectToAddress(PlainSocketImpl.java:213)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:200)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:366)
	at java.net.Socket.connect(Socket.java:529)
	at java.net.Socket.connect(Socket.java:478)
	at java.net.Socket.<init>(Socket.java:375)
	at java.net.Socket.<init>(Socket.java:249)
	at org.apache.commons.httpclient.protocol.DefaultProtocolSocketFactory.createSocket(DefaultProtocolSocketFactory.java:79)
	at org.apache.commons.httpclient.protocol.DefaultProtocolSocketFactory.createSocket(DefaultProtocolSocketFactory.java:121)
	at org.apache.commons.httpclient.HttpConnection.open(HttpConnection.java:706)
	at org.apache.commons.httpclient.HttpMethodDirector.executeWithRetry(HttpMethodDirector.java:386)
	at org.apache.commons.httpclient.HttpMethodDirector.executeMethod(HttpMethodDirector.java:170)
	at org.apache.commons.httpclient.HttpClient.executeMethod(HttpClient.java:396)
	at org.apache.commons.httpclient.HttpClient.executeMethod(HttpClient.java:324)
	at com.mirth.connect.connectors.ws.WebServiceDispatcher.getWsdlUrl(WebServiceDispatcher.java:163)
	at com.mirth.connect.connectors.ws.WebServiceDispatcher.createDispatch(WebServiceDispatcher.java:132)
	at com.mirth.connect.connectors.ws.WebServiceDispatcher.send(WebServiceDispatcher.java:228)
	at com.mirth.connect.donkey.server.channel.DestinationConnector.handleSend(DestinationConnector.java:612)
	at com.mirth.connect.donkey.server.channel.DestinationConnector.run(DestinationConnector.java:517)
	at java.lang.Thread.run(Thread.java:662)

{noformat} 

Imported Issue. Original Details:
Jira Issue Key: MIRTH-3458
Reporter: piet
Created: 2014-09-16T00:57:27.000-0700

**Comment by rbeckman-nextgen** [comment]
Connection errors are handled while making the actual call, but not creating the dispatch.

I think adding handling of connection errors in the outer try/catch block should solve the problem.


Imported Comment. Original Details:
Author: piet
Created: 2014-09-16T01:01:10.000-0700

**Comment by rbeckman-nextgen** [comment]
Revision 7373: Fixed bug that caused Web Service Sender dispatches to not queue if an error occurred creating the Dispatch object. This could happen due to connectivity problems when pulling down the WSDL, or it could happen if the service/port are not correct. Instead of setting the status to ERROR, we'll just leave it as QUEUED in these cases. If a message gets queued because the service or port are incorrect, fixing it in the connector properties and enabling Regenerate Template will resolve the issue.

Imported Comment. Original Details:
Author: narupley
Created: 2014-09-16T12:38:01.000-0700

**Comment by rbeckman-nextgen** [comment]
Created a Web service listener channel and a web service sender channel in 3.0.3. I had queuing on my sending channel set to attempt first and tried to dispatch a message to my listener which was stopped. The message in this case error-ed out! I performed this same test with a channel reader and writer and verified this messaged queued up.

I exported these channels to 3.1 and attempted the same steps. This time, the channel correctly queued the message. However, when I did the same for the channel writer/reader, the message was queued and an error was logged on the dashboard to inform me the source connector was stopped. Not sure if we would like to provide an error message within the dashboard but this also occurs for other connectors such as TCP. 

Imported Comment. Original Details:
Author: jaysenp
Created: 2014-09-17T17:56:42.000-0700
